### PR TITLE
[FIX] base_vat: only check vat in case we check the commercial partner

### PR DESCRIPTION
**Pull request by wtaferner**
**Description of the issue/feature this PR addresses:**

Follow-up of https://github.com/odoo/odoo/pull/42973

@nim-odoo @mart-e @qdp-odoo 
Probably easier to forward port and to digest in the end...

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by wtaferner** [comment]
@nim-odoo 
May I ask you for a review? It is working as expected IMO.

**Comment by mart-e** [comment]
@wtaferner I have added a test, do you have valid test VAT number or are you fine with this one? ;-)
I am not sure this is a good idea anyway. The `vat` field is readonly for contacts but it's only in the interface, are we sure we want to disable the vat check for such scenario?

@nim-odoo what do you think? Do you see cases where it's problematic?

I have made an alternative patch at odoo-dev/odoo@4e1de7e59ec8e3cb0ad26eb7e7e260c5c6b6a28e that may be safer, to test

**Comment by wtaferner** [comment]
@mart-e 
Thank you for adding a test. I am feeling honored but I think BE 0477472701 would be better ;-)
The use case is pretty straight forward IMO. 

As it is automatically checking the contact now (redundant), even though it is just a copy of the commercial field, it is clear that a contact should never have an own or different VAT number as their commercial partner except it is a company itself and therefore also its own commercial partner (pass condition). 

At least this is how Odoo is implemented. In reality it is easily possible that a company has other addresses and VAT numbers in other EU countries and therefore have multiple VAT numbers, but this is out of scope for the default commercial partner scenario configuration.

Your fix seems to be valid too and maybe "safer", but it will do an extra check for every contact to VIES (ok, I think there was a fix which is now caching results, but why check a copied field anyway)

File: addons/base_vat/tests/test_validate_ruc.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 from odoo.exceptions import ValidationError
 
 
-class TestRUCStructure(common.TransactionCase):
+class TestStructure(common.TransactionCase):
```
Comment (nim-odoo) [inline addons/base_vat/tests/test_validate_ruc.py]: The file name was intended for PE validation only. Maybe the file needs to be renamed as well?

Comment (mart-e) [reply addons/base_vat/tests/test_validate_ruc.py]: I just tired but the file changes too much and git can not detect the renaming so we lose the blame on `test_peru_ruc_format`.

**Comment by mart-e** [comment]
> Result is indeed cached in #44260

Oh this seems to break my test ^^

> The VAT number [BE0123456789] for partner [World Company] does not seem to be valid.

**Pull request closed without merging**